### DISCUSSION TITLE
VS Code Extension

### DISCUSSION THREAD

**Discussion by itavero** [cpm-cmake/CPM.cmake#470] [Show and tell]
At work we have also been using CPM.cmake and we were already using VS Code a lot with the CMake Tools extension.

One of the "complaints" I got, was that it wasn't always easy to find the contents of a package pulled in via CPM.cmake since we are using a global source cache (via the `CPM_SOURCE_CACHE` environment variable).
So, I decided to "hack" together an VS Code extension myself: [CPM.cmake Manager](https://marketplace.visualstudio.com/items?itemName=arno-dev.cpm-cmake-manager).

When a `CMakeList.txt` is detected in the workspace (which should also trigger the CMake Tools extension), it tries to figure out where the `CMakeCache.txt` is and extracts the CPM related stuff from it.
Based on that information, it shows a list of included packages in the Explorer view of VS Code.
From that list you can easily open a new VS Code window for the source directory of a package and see which version is used (if it's a Git dependency, it also tries to see if there are newer tags available and show you that information as well).

Just a first version and there's probably room for lots of improvements, but I thought I'd share this as it might be of use to others as well.

**Comment by TheLartians** [comment]
That's awesome, thanks for sharing! 

Out of curiosity, you planning to publish the source code of the extension (the provided [GitHub URL](https://github.com/itavero/vscode-cpm-cmake-manager) seems to lead nowhere)?

I think it would be great to add a section for tools like this to the Readme to allow easier discoverability by users.

**Reply by itavero** [reply]
Oops! Forgot to change the visibility of the repository. It should work now 😅
